src/Symfony/Component/Security/Http/AccessToken/AccessTokenHandlerInterface.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\AccessToken;
+
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+
+/**
+ * The token handler retrieves the user identifier using the token.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ */
+interface AccessTokenHandlerInterface
+{
+    /**
+     * @throws AuthenticationException
+     */
+    public function getUserIdentifierFrom(string $accessToken): string;
+}
```

Comment:
Do we really need this abstraction? 

I mean, we can use the regular way to fetch a user thanks to the UserProvider.

```php
class AccessTokenAuthenticator implements AuthenticatorInterface
{
    public function authenticate(Request $request): Passport
    {
        $accessToken = $this->accessTokenExtractor->extractAccessToken($request);
        if (null === $accessToken) {
            throw new BadCredentialsException('Invalid credentials.');
        }

        return new Passport(
            new UserBadge($accessToken, $this->userProvider->loadUserByIdentifier(...)),
            new AccessTokenCredentials($accessToken)
        );
    }
```

Then, if some extra steps are needed, it can be done in the UserProvider and / or in the SuccessHandler or via a listener

cc @wouterj

Reply:
We don't know how the access token is computed. The user ID can be a JWT custom claim `user_id` or `sub` or a or column in a DB. This method is used to anwser he question: what is the user ID linked to this access token.
Then the UserProvider is called to retreive the user object.

Reply:
Yes, I got this part. But to me it can be simpler.

https://github.com/symfony/symfony/blob/f42339552cdeee8aa1e6286c2c97dff1cca3032e/src/Symfony/Component/Security/Core/User/UserProviderInterface.php#L57-L63 told us:

> Loads the user for the given user identifier (e.g. username or email).


A JWT **is** an identifier (yes, it contains more information, but still).

So all the code you put in the "PR descriptions > examples" can live in a UserProvider IMHO.

Reply:
Yes, but the user provider is just supposed to return a user for a given identifier.
Here, the method should also validate the access token. I mentioned it in https://github.com/symfony/symfony/pull/46428#issue-1244016678, but should also be mentioned in the phpdoc

Reply:
I see your point but I disagree. The token can indeed be the user identifier, but it's not always the case and my guess it won't be for most usages of this feature. One doesn't want to have the token loading/decoding+validation+user id extraction logic in the user provider as it would go way too much above UserProviderInterface's scope. This single abstraction feels structuring and is the only thing you need to implement as a bundle providing e.g.  a JWT authenticator.

Reply:
However, I think that at least the interface phpdoc could be extended to give more hints about what should be done by implementations. I say "at least" because if it's likely to perform some validation then `getUserIdentifierFrom()` might not be the right method name.

Reply:
OKay OKay. I could use the very same class anyway for both method 👍🏼